numpy/distutils/from_template.py

```diff
@@ -83,8 +83,8 @@ def parse_structure(astr):
     return spanlist
 
 template_re = re.compile(r"<\s*(\w[\w\d]*)\s*>")
-named_re = re.compile(r"<\s*(\w[\w\d]*)\s*=\s*(.*?)\s*>")
-list_re = re.compile(r"<\s*((.*?))\s*>")
+named_re = re.compile(r"<\s*(\w[\w\d]*)\s*=\s*(\S*?)\s*>")
+list_re = re.compile(r"<\s*((\S*?))\s*>")
```

Comment:
These two changes look correct, but I don't understand why they makes any difference.

Reply:
This sub-pattern `\s*(\S*?)\s*` will still produce catastrophic backtracking. 
i think you can replace `\s*(\S*?)\s*` with `\s*(\S+\s*)?`

Reply:
Thanks @yetingli yes that makes sense, I just updated my PR and also added comments using re.VERBOSE.  The goal is to make the require less backtracking and include the termination within the regex as it moves forward instead of matching arbitrarily long strings and then finding a mismatch afterwards that requires backtracking.